src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -239,40 +239,53 @@ private BoundStatement BindYieldReturnStatement(YieldStatementSyntax node, Bindi
                 ? BadExpression(node).MakeCompilerGenerated()
                 : BindValue(node.Expression, diagnostics, BindValueKind.RValue);
 
-            if (!argument.HasAnyErrors)
-            {
-                argument = GenerateConversionForAssignment(elementType, argument, diagnostics);
-            }
-            else
-            {
-                argument = BindToTypeForErrorRecovery(argument);
-            }
+            bool hasErrors = false;
 
             // NOTE: it's possible that more than one of these conditions is satisfied and that
             // we won't report the syntactically innermost.  However, dev11 appears to check
             // them in this order, regardless of syntactic nesting (StatementBinder::bindYield).
             if (this.Flags.Includes(BinderFlags.InFinallyBlock))
             {
                 Error(diagnostics, ErrorCode.ERR_BadYieldInFinally, node.YieldKeyword);
+                hasErrors = true;
```

Comment:
>hasErrors = true; [](http://example.com/codeflow?start=16&length=17)

Why would we want to disable conversion binding for this and the other modified code paths. It feels to me that the conversion checks and the "unsupported" context checks are orthogonal, therefore cannot be considered as cascading for one another

Reply:
I woke up this morning with the same thought, but you'd already reviewed. Thanks